lib/multi-select.js

```diff
@@ -1,232 +1,115 @@
-   */
-  prepare (row, col, prop, td, originalValue, cellProperties) {
-    super.prepare(row, col, prop, td, originalValue, cellProperties)
-    const { type } = cellProperties
-    this.type = type
-
-    const selectOptions = R.prop('select', cellProperties)
-    this.selectOptions = R.defaultTo({}, selectOptions)
-
-    this.valueKey = getValueKey(this.selectOptions) || 'value'
-    this.labelKey = getLabelKey(this.selectOptions) || 'label'
-
-    if (type === 'numeric') {
-      this.separator = '|'
-      this.maxItemCount = 1
-      this.originalValue = isNumber(originalValue) ? originalValue.toString() : originalValue
-    } else {
-      this.separator = getSeparator(this.selectOptions) || ','
-      this.maxItemCount = getMaxItemCount(this.selectOptions) || -1
-    }
-  }
+  getValue = () => this.choices.getValue().join(this.separator)
 
   /**
    * Creates an editor's elements and adds necessary CSS classnames.
    */
   createElements () {
+    // Create textarea and parent
+    this.TEXTAREA_PARENT = document.createElement('div')
     this.TEXTAREA = document.createElement('select')
-    this.TEXTAREA.select = () => {}
-    this.TEXTAREA.tabIndex = -1
+    this.TEXTAREA.select = () => null
+    this.TEXTAREA.style.display = 'none'
     addClass(this.TEXTAREA, 'handsontableInput')
-    this.TEXTAREA.setAttribute('multiple', true)
-
-    this.TEXTAREA_PARENT = document.createElement('div')
     addClass(this.TEXTAREA_PARENT, 'handsontableInputHolder')
-
-    if (hasClass(this.TEXTAREA_PARENT, this.layerClass)) {
-      removeClass(this.TEXTAREA_PARENT, this.layerClass)
-    }
-
     addClass(this.TEXTAREA_PARENT, EDITOR_HIDDEN_CLASS_NAME)
 
-    this.textareaStyle = this.TEXTAREA.style
-    this.textareaStyle.width = 0
-    this.textareaStyle.height = 0
-    this.textareaStyle.overflowY = 'visible'
-
     this.textareaParentStyle = this.TEXTAREA_PARENT.style
 
     this.TEXTAREA_PARENT.appendChild(this.TEXTAREA)
     this.instance.rootElement.appendChild(this.TEXTAREA_PARENT)
-  }
 
-  /**
-   * Opens the editor and adjust its size.
-   */
-  open () {
-    this.refreshDimensions()
-    this.showEditableElement()
+    this.choices = new ChoicesDropdown(this.TEXTAREA)
+    this.choices.renderStructure()
+
+    this.choices.addEventListener('explicitKeydown', ({ event }) => this.onBeforeKeyDown(event))
+    this.choices.addEventListener('change', ({ isMultiple, actionType }) => {
+      if (!isMultiple && actionType === 'add' && this.isOpened()) {
+        this.instance.destroyEditor()
+      }
+    })
 
     this.instance.addHook('beforeKeyDown', (event) => this.onBeforeKeyDown(event))
+    this.TEXTAREA.addEventListener('hideDropdown', this.close.bind(this))
+  }
 
-    if (this.choices) this.choices.destroy()
+  /**
+   * Prepares editor's meta data.
+   *
+   * @param {options} args
+   */
+  prepare (...args) {
+    super.prepare(...args)
+    const [ , , , , originalValue, cellProperties ] = args
```

Comment:
you can do:

```
prepare (row, col, prop, td, originalValue, cellProperties) {
  super.prepare({ row, col, prop, td, originalValue, cellProperties })
}
```

Reply:
My way feels cleaner. There are no duplications. Plus, we won't need to change this line if the original prepare function changes its arguments model.